### PR TITLE
feat: forbid generating an access token without a session

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Enforces the precondition of a valid session before one can create an access token.  This supports refactors around `generateAccessToken` and  `updateMFASessionAndClaims`. Also allows for stronger guarantees within the function since one can always assume there is a valid session.

There were a few test changes:
- To mirror real world use, Access Tokens should now only exist where there is a valid session. We wrap  `generateAccessToken` into a helper `generateAccessTokenAndSession` to replace previous occurrences where session was set to nil.
- We split TestUpdatePassword into cases where reauthentication is required and reauthentication is not required. We also attach a session to two of the test cases as they were previously nil

### CODE REVIEW

File: internal/api/phone_test.go [hunk 1]
```diff
@@ -258,3 +257,14 @@ func (ts *PhoneTestSuite) TestMissingSmsProviderConfig() {
 		}
 	}
 }
+
+func (ts *PhoneTestSuite) generateAccessTokenAndSession(ctx context.Context, u *models.User, authenticationMethod models.AuthenticationMethod) string {
```
Comment (J0) [inline internal/api/phone_test.go:261]: It would be great if this could be a global static method though I'm not sure if there's currently a way to do so

Comment (kangmingtay) [reply internal/api/phone_test.go:261]: yeah, not too sure if it's worth abstracting this into a function since we're only using it in one place in this file 

nit: if you opt to keep it, can we rename it to `generateAccessTokenString` since the function only returns the access token and not the session?

**Review by kangmingtay — APPROVED** [review]

File: internal/api/phone_test.go (hunk 1 above)
Comment (J0) [reply internal/api/phone_test.go:261]: Will drop it
